pkg/cloud-provider/cloudapi/azure/azure_security_test.go

```diff
@@ -0,0 +1,404 @@
+
+			vmIDToInfoMap[cloudcommon.InstanceID(testVMID01)] = &virtualMachineTable{
+				VnetID: &testVnetPeerID01,
+				NetworkInterfaces: []*networkInterface{
+					{
+						PrivateIps: []*string{
+							&testPrivateIP,
+						},
+					},
+				},
+			}
+			serviceConfig.(*computeServiceConfig).resourcesCache.UpdateSnapshot(&computeResourcesCacheSnapshot{vmIDToInfoMap, vnetIDs, vpcPeers})
+		})
+
+		AfterEach(func() {
+			mockCtrl.Finish()
+		})
+
+		Context("CreateSecurityGroup", func() {
+			It("Should create security group(ASG and NSG) successfully and return ID", func() {
+				webAddressGroupIdentifier01 := &securitygroup.CloudResourceID{
+					Name: "Web",
+					Vpc:  testVnetID01,
+				}
+
+				cloudSgID01, err := c.CreateSecurityGroup(webAddressGroupIdentifier01, false)
+				Expect(err).Should(BeNil())
+				Expect(cloudSgID01).Should(Not(BeNil()))
+
+				webAddressGroupIdentifier02 := &securitygroup.CloudResourceID{
+					Name: "Web",
+					Vpc:  testVnetID02,
+				}
+
+				cloudSgID02, err := c.CreateSecurityGroup(webAddressGroupIdentifier02, true)
+				Expect(err).Should(BeNil())
+				Expect(cloudSgID02).Should(Not(BeNil()))
+
+			})
+		})
+
+		Context("UpdateSecurityGroup", func() {
+			It("Should create security group(ASG and NSG) successfully and return ID", func() {
+				webAddressGroupIdentifier01 := &securitygroup.CloudResourceID{
+					Name: "Web",
+					Vpc:  testVnetID01,
+				}
+
+				webAddressGroupIdentifier02 := &securitygroup.CloudResourceID{
+					Name: "Web",
+					Vpc:  testVnetID02,
+				}
+
+				members := []*securitygroup.CloudResource{
+					{
+						Type: securitygroup.CloudResourceTypeVM,
+						Name: *webAddressGroupIdentifier02,
+					},
+				}
+
+				err := c.UpdateSecurityGroupMembers(webAddressGroupIdentifier01, members, false)
+				Expect(err).Should(BeNil())
+			})
+		})
+
+		Context("UpdateSecurityRules", func() {
+			It("Should update Security rules", func() {
+				webAddressGroupIdentifier03 := &securitygroup.CloudResourceID{
+					Name: atAsgName,
+					Vpc:  testVnetID01,
+				}
+
+				cidr := ipaddr.NewIPAddressString(testCidrStr)
+				subnet, _ := cidr.GetAddress().ToPrefixBlock(), cidr.GetHostAddress()
+				var ipNet = net.IPNet{
+					IP:   subnet.GetNetIP(),
+					Mask: subnet.GetNetworkMask().Bytes(),
+				}
+				fromSrcIP := []*net.IPNet{
+					&ipNet,
+				}
+
+				ingressrules := []*securitygroup.IngressRule{
+					{
+						Protocol:  &testProtocol,
+						FromPort:  &testFromPort,
+						FromSrcIP: fromSrcIP,
+					},
+				}
+
+				egressrules := []*securitygroup.EgressRule{
+					{
+						Protocol: &testProtocol,
+						ToPort:   &testToPort,
+						ToDstIP:  fromSrcIP,
+						ToSecurityGroups: []*securitygroup.CloudResourceID{
+							webAddressGroupIdentifier03,
+						},
+					},
+				}
+
+				err := c.UpdateSecurityGroupRules(webAddressGroupIdentifier03, ingressrules, egressrules)
+				Expect(err).Should(BeNil())
+			})
+
+			It("Should update Security rules for Peerings", func() {
```

Comment:
For now, remove peering test. We will add it when some coverage on integration tests too

Reply:
deleted